### DISCUSSION TITLE
[Node Operator Question] RPC network error

### DISCUSSION THREAD

**Discussion by makarid** [ethereum-optimism/developers#195] [Node Operators] [answered]
### Are you running the most up to date node software?

- [X] Yes

### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hello, i used the guide on the documentation and installed Node from source. Everything went well, i can see that the logs are very similar to the logs that i must see on the Synchronization section of the guide. When i am calling:

curl -X POST -H "Content-Type: application/json" --data \
    '{"jsonrpc":"2.0","method":"optimism_syncStatus","params":[],"id":1}'  \
    http://localhost:8647

on my terminal, i am getting a response. But when i am calling this code from my js file:

const provider1 = new ethers.providers.JsonRpcProvider("http://localhost:8647")
console.log(await provider1.getBalance("address"))

i am getting "Could not detect network".

When i am running the following code:

const provider1 = new ethers.providers.JsonRpcProvider("http://localhost:8647" , 'op-mainnet')
console.log(await provider1.getBalance("address"))

i am getting "Invalid network error"

Also, websocket on port 8646 is not working either.

I am running Stereum on the same machine for Ethereum mainnet.
Is this error possible because op-node hasn't been 100% synced? 
What else i need to search in order to know that RPC is working or is not working correctly (any commands i can run or something).

So far i have tried to change port numbers, allow all local network ports from ufw. I have run the same command using QuickNode RPC and it worked without any issue.
I am using EtherJs V5.

My goal is to use my local RPC instead of online node providers.

Thank you very much for your time.

### Protocol Description

op-geth v1.101305.0
op-node v1.4.2
network: op-mainnet

### Additional Information

My config for op-geth:

         --port=30403 \
         --ws \
         --ws.port=8646 \
         --ws.addr=localhost \
         --ws.origins="*" \
         --http \
         --http.port=8645 \
         --http.addr=localhost \
         --http.vhosts="*" \
         --http.corsdomain="*" \
         --authrpc.addr=localhost \
         --authrpc.port=8551 \
         --authrpc.vhosts="*" \
         --authrpc.jwtsecret=./jwt.txt \
         --verbosity=3 \
         --rollup.sequencerhttp=https://mainnet-sequencer.optimism.io/ \
         --nodiscover \
         --syncmode=full \
         --maxpeers=0 \
         --op-network=op-mainnet \
         --datadir=<extracted_snapshot_dir>

My config for op-node:

  --l1= QuickNode API \
  --l1.rpckind=quicknode \
  --l2=ws://localhost:8551 \
  --l2.jwt-secret=./jwt.txt \
  --network=op-mainnet \
  --rpc.addr=localhost \
  --rpc.port=8647 \
  --metrics.enabled

**Comment by opfocus** [comment]
If I'm not mistaken, "optimism_syncStatus" is a request sent to the op-node. Please confirm if your op-geth's port is also 8647.    Please try 8646/8645

**Reply by makarid** [reply]
I am getting 'could not detect network' error.

**Reply by opfocus** [reply]
I think it is appropriate to use const provider1 = new ethers.providers.JsonRpcProvider("http://localhost:8645") here, but I'm not sure about the reason for getting the 'could not detect network' error. Some RPC queries may not need to wait for synchronization to complete; perhaps you can try some queries for op-geth mentioned here: https://docs.alchemy.com/reference/eth-blocknumber-optimism

**Comment by sbvegan** [comment]
Yeah @opfocus is right here, the `optimism_*` rpc methods are for the `op-node`. It seems like the port is right. For context, the OP Stack is very similar to Ethereum, where the client is broken into the execution layer and the consensus layer. For OP Mainnet that is `op-geth` (execution) and `op-node` (consensus).

Now in your JS code, you're calling `getBalance`, but I believe this call should be for the `op-geth` provider at port: `8646`.

**Reply by avenbreaks** [reply]
thank you for your answer, I'm from a different project but wanted to ask for an answer about the same thing. 

in the op-geth `-rollup.sequencerhttp= <seq_rpc>` section where should this be pointed? does this have to be filled in from the op-node? and sorry I'm confused with the sequencer. 

`op-geth (execution) `
`op-node (consensus)` then the sequencer is from this op-node client?

**Reply by sbvegan** [reply, marked as answer]
[--rollup.sequencerhttp](https://docs.optimism.io/builders/node-operators/management/configuration#rollupsequencerhttp): HTTP endpoint of the sequencer. op-geth will route eth_sendRawTransaction calls to this URL.

For OP Mainnet, its `https://mainnet-sequencer.optimism.io` - [source](https://docs.optimism.io/chain/networks)

The sequencer is a privileged node on the network, its like the rest of the rollup nodes, but it works with the batcher and proposer to build the L2 blockchain. These privileged services are allowed to make new blocks. Similar to Ethereum L1, the rollup node is made of the a execution and consensus client. `op-geth` + `op-node` = `rollup node`
